src/lib_info.ml

```diff
@@ -84,6 +85,12 @@ let user_written_deps t =
 let of_library_stanza ~dir ~has_native ~ext_lib ~ext_obj
       (conf : Dune_file.Library.t) =
   let (_loc, lib_name) = conf.name in
+  let orig_src_dir =
+    match Path.drop_build_context dir with
+    | None -> None
+    | Some src_dir ->
+      Some Path.(of_string (to_absolute_filename (in_source (to_string src_dir))))
```

Comment:
This is weird and maybe can be done easier but `Path` is confusing for me at the moment.

Reply:
`in_source (to_string src_dir)` should be the identity, you can replace it by `src_dir`